File: aula3/Personagem.java
// Projeto IF Quest
// Aula 3 - POO


// Classe com atributos e comportamentos de cada pesonagem
public class Personagem {
    private String nome;
    private int vida;
    private int nivel;

    public Personagem(String nome) {
        definirNome(nome);
        definirVida(100);
        definirNivel(1);
    }

    public void definirNome(String nome) {
        this.nome = nome;
    }

    public String obterNome() {
        return this.nome;
    }
    
    public void definirVida(int vida) {
        if (vida < 0) {
            this.vida = 0;
        } else if (vida > 100) {
            this.vida = 100;
        } else {
            this.vida = vida;
        }
    }

    public int obterVida() {
        return this.vida;
    }

    public void definirNivel(int nivel) {
        if (nivel < 1) {
            this.nivel = 1;
        } else {
            this.nivel = nivel;
        }
    }

    public int obterNivel() {
        return this.nivel;
    }


}



// Classe principal
class Main {
    public static void main(String[] args) {
        // Criando o personagem
        Personagem teste = new Personagem("Scrible");
        
        // Testando os métodos getter
        System.out.println("Nome: " + teste.obterNome());
        System.out.println("Vida: " + teste.obterVida());
        System.out.println("Nível: " + teste.obterNivel());

        System.out.println("\nAlterando os valores dos atributos...\n");
        
        // Testando os métodos setter
        teste.definirNome("Scrible 2");
        teste.definirVida(150);
        teste.definirNivel(0);

        // Testando os métodos getter novamente
        System.out.println("Nome: " + teste.obterNome());
        System.out.println("Vida: " + teste.obterVida());
        System.out.println("Nível: " + teste.obterNivel());

        // Testando a alteração de vida para um valor negativo
        teste.definirVida(-50);

        // Testando os métodos getter novamente após alteração
        System.out.println("\nApós tentar alterar a vida para um valor negativo:");
        System.out.println("Nome: " + teste.obterNome());
        System.out.println("Vida: " + teste.obterVida());
        System.out.println("Nível: " + teste.obterNivel());
    }
}

File: aula3/Personagem.py
# Projeto IF Quest
# Aula 3 - POO


# Classe com atributos e comportamentos de cada pesonagem
class Personagem:
    def __init__(self, nome):
        self._nome = nome
        self._vida = 100
        self._nivel = 1
    
    @property
    def nome(self):
        return self._nome

    @nome.setter
    def nome(self, nome):
        self._nome = nome

    @property
    def vida(self):
        return self._vida

    @vida.setter
    def vida(self, vida):
        if (vida < 0):
            self._vida = 0
        elif (vida > 100):
            self._vida = 100
        else:
            self._vida = vida
        
    
    @property
    def nivel(self):
        return self._nivel
    

    @nivel.setter
    def nivel(self, nivel):
        if (nivel < 1):
            self._nivel = 1
        else:
            self._nivel = nivel
        



# Criando o personagem
teste = Personagem("Scrible")

# Testando o getter
print("Nome: ", teste.nome)
print("Vida: ", teste.vida)
print("Nivel: ", teste.nivel)

print("\nAlterando os valores dos atributos...\n")

# Testando o setter
teste.nome = "Scrible 2"
teste.vida = 150
teste.nivel = 0

# Testando novamente os getters após alteração
print("Nome: ", teste.nome)
print("Vida: ", teste.vida)
print("Nivel: ", teste.nivel)

# Testando a alteração de vida para um valor negativo
teste.vida = -50

# Testando novamente os getters após alteração 
print("\nApós tentar alterar a vida para um valor negativo:")
print("Nome: ", teste.nome)
print("Vida: ", teste.vida)
print("Nivel: ", teste.nivel)